README.md

```diff
@@ -365,49 +390,43 @@ negate([
+The Z1 operator returns the previous input [Z-Set].
 
 ## 2.9.11 Distinct Trace Operator
 
-(TODO Like with Join Stream and Join Trace, it may make more sense to specify this alongside the Distinct operator as suggestions for incrementalizing it).
+Distinct trace ia a variant of [Distinct] that offers more performance for incremental computation and computes across multiple timestamps, with support for use in nested contexts, like recursive circuits. It computes the distinct elements of a [Z-Set] in its first argument, with respect to a [Trace] in its second, returning them in a new $\Bbb{Z}$-Set. The resulting $\Bbb{Z}$-Set has no timestamps associated with any element.
 
-A variant of [Distinct](#293-distinct-operator) that offers more performance for incremental computation and computes across multiple timestamps, with support for use in nested contexts, like recursive circuits. It computes the distinct elements of a [ZSet](#21-zset) in its first argument, with respect to a [Trace](#23-trace) in its second, returning them in a new ZSet. The resulting ZSet has no timestamps associated with any element.
-
-Note that because operator computes the delta of [Distinct](#293-distinct-operator), it is possible for returned elements to have negative weights, if those elements are deleted between timestamps.
+Note that because operator computes the delta of [Distinct], it is possible for returned elements to have negative weights, if those elements are deleted between timestamps.
 
 Distinct is not a linear operation, and requires access to the entire history of updates, however there's a few observations that can reduce the search space of timestamps to examine.
 
-Consider evaluating the operator at some [timestamp](#25-time) `t = (e, i)`, where `e` denotes the epoch, and `i` denotes the iteration.
+Consider evaluating the operator at some [timestamp] $t = \langle e, i \rangle$, where $e$ denotes the epoch, and $i$ denotes the iteration.
 
 Then there are two possible classes of elements which may be returned:
-1) Elements in the current input ZSet
-2) Elements that were returned in response to a previous input, at timestamp `(e, i0)`, such that `i0 < i`, and where the element was also returned at timestamp `(e0, i)`, such that `e0 < e`
-
-For each element, with weight `w`, meeting at least one of the above requirements, if that element does not appear in the trace, it is returned in the ZSet with weight 1. Otherwise, the following routine is performed:
-
-1) `w1` is computed, as the sum of all weights in which that element appears at times `(e0, i0)` for `e0 < e` and `i0 < i`
-2) `w2` is computed, as the sum of all weights in which that element appears at times `(e0, i)` for `e0 < e`
-3) `w3` is computed, as the sum of all weights in which that element appears at times `(e, i0)` for `i0 < i`
-4) `d0` is computed, such that:
-   1) `d0 = 1`, if `w1 <= = && w1 + w2 > 0`
```

Comment:
I'm guessing that the `=` here is a typo, not the `<=`, ya?

Reply:
Good catch! It should be `w1 <= 0 and w1 + w2 > 0`.

For completeness, here's my Elixir implementation:
```elixir
# Sum of weights where epoch(t) < epoch(time) and iteration(t) < iteration(time)
w1 = 0

# Sum of weights where epoch(t) < epoch(time) and iteration(t) == iteration(time)
w2 = 0

# Sum of weights where epoch(t) == epoch(time) and iteration(t) < iteration(time)
w3 = 0

{trace_cursor, times} = ZSetTraceCursor.times(trace_cursor)

{w1, w2, w3, next_time} =
  Enum.reduce(times, {w1, w2, w3, nil}, fn
    {{t_epoch, t_iteration} = t, w}, {w1, w2, w3, next_time} ->
      cond do
        t_epoch == epoch ->
          if Algebra.PartialOrder.less_than(t_iteration, iteration) do
            {w1, w2, w3 + w, next_time}
          else
            {w1, w2, w3, next_time}
          end

        Algebra.PartialOrder.less_than(t_iteration, iteration) ->
          {w1 + w, w2, w3, next_time}

        t_iteration == iteration ->
          {w1, w2 + w, w3, next_time}

        is_nil(next_time) or Algebra.PartialOrder.less_than(t, next_time) ->
          {w1, w2, w3, t}

        true ->
          {w1, w2, w3, next_time}
      end
  end)

w12 = w1 + w2
w13 = w1 + w3
w123 = w12 + w3
w1234 = w123 + weight

delta_old =
  cond do
    w1 <= 0 and w12 > 0 ->
      1

    w1 > 0 and w12 <= 0 ->
      -1

    true ->
      0
  end

delta_new =
  cond do
    w13 <= 0 and w1234 > 0 ->
      1

    w13 > 0 and w1234 <= 0 ->
      -1

    true ->
      0
  end

output =
  if delta_old == delta_new do
    output
  else
    w = delta_new - delta_old
    tuple = {{key, {}}, nil, w}

    [tuple | output]
  end
  ```